pkg/operation/botanist/component/etcd/etcd.go

```diff
@@ -461,7 +461,7 @@ func (e *etcd) Deploy(ctx context.Context) error {
 		}
 
 		// TODO(timuthy): Once https://github.com/gardener/etcd-backup-restore/issues/538 is resolved we can enable PeerUrlTLS for all remaining clusters as well.
-		if pointer.Int32Deref(e.values.Replicas, 0) > 1 {
+		if pointer.Int32Deref(e.values.Replicas, 0) > 1 || (existingEtcd != nil && existingEtcd.Spec.Etcd.PeerUrlTLS != nil) {
```

Comment:
I'd recommend to use a self expressive var that you can (re)use in the code like:
```
peerTLSAlreadyEnabled := existingEtcd != nil && existingEtcd.Spec.Etcd.PeerUrlTLS != nil
```